sdk/storage/inc/datalake/file_client.hpp

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "common/storage_credential.hpp"
+#include "common/storage_url_builder.hpp"
+#include "datalake/path_client.hpp"
+#include "datalake_options.hpp"
+#include "http/pipeline.hpp"
+#include "protocol/datalake_rest_client.hpp"
+
+#include <memory>
+#include <string>
+
+namespace Azure { namespace Storage { namespace Files { namespace DataLake {
+
+  struct ReadPathResponse
+  {
+    std::unique_ptr<Azure::Core::Http::BodyStream> Body;
+    std::string AcceptRanges;
+    DataLakeHttpHeaders HttpHeaders;
+    int64_t ContentLength = int64_t();
+    Azure::Core::Nullable<int64_t> RangeOffset;
+    Azure::Core::Nullable<int64_t> RangeLength;
+    Azure::Core::Nullable<std::string> TransactionalMD5;
+    std::string Date;
+    std::string ETag;
+    std::string LastModified;
+    std::string RequestId;
+    std::string Version;
+    std::string ResourceType;
+    Azure::Core::Nullable<std::string> LeaseDuration;
+    LeaseStateType LeaseState;
+    LeaseStatusType LeaseStatus;
+    Azure::Core::Nullable<std::string> ContentMD5;
+    std::map<std::string, std::string> Metadata;
+  };
+
+  struct FileRenameResponse
+  {
+    std::string Date;
+    Azure::Core::Nullable<std::string> ETag;
+    Azure::Core::Nullable<std::string> LastModified;
+    std::string RequestId;
+    std::string Version;
+  };
+
+  struct FileDeleteResponse
+  {
+    std::string Date;
+    std::string RequestId;
+    std::string Version;
+  };
+
+  using FileInfo = PathInfo;
+  using FileCreateOptions = PathCreateOptions;
+
+  class FileClient : public PathClient {
+  public:
+    /**
+     * @brief Create from connection string
+     * @param connectionString Azure Storage connection string.
+     * @param fileSystemName The name of a file system.
+     * @param filePath The path of a file within the file system.
+     * @param options Optional parameters used to initialize the client.
+     * @return FileClient
+     */
+    static FileClient CreateFromConnectionString(
+        const std::string& connectionString,
+        const std::string& fileSystemName,
+        const std::string& filePath,
+        const FileClientOptions& options = FileClientOptions());
+
+    /**
+     * @brief Shared key authentication client.
+     * @param pathUri The URI of the file this client's request targets.
+     * @param credential The shared key credential used to initialize the client.
+     * @param options Optional parameters used to initialize the client.
+     */
+    explicit FileClient(
+        const std::string& filePathUri,
```

Comment:
Why not follow the same pattern, `fileUri`?

Reply:
I was thinking if we should point out that the URI needs to contain the path of the file as well in the parameter naming. But I think `fileUri` is better.